lib/nx_gzlib.c

```diff
@@ -82,37 +92,46 @@ int nx_gzclose(gzFile file)
 			stream->next_out = next_out;
 			stream->avail_out = MAX_LEN;
 
-			ret = nx_deflate(stream, Z_FINISH);
+			if (force_nx)
+				ret = nx_deflate(stream, Z_FINISH);
+			else
+				ret = deflate(stream, Z_FINISH);
+
 			w_ret = write(state->fd, next_out,
 				      MAX_LEN - stream->avail_out);
 		} while (ret != Z_STREAM_END && w_ret >= 0);
 
-		ret = nx_deflateEnd(stream);
-	} else
-		ret = nx_inflateEnd(stream);
+		if (force_nx)
+			ret = nx_deflateEnd(stream);
+		else
+			ret = deflateEnd(stream);
+
+	} else {
+		if (force_nx)
+			ret = nx_inflateEnd(stream);
+		else
+			ret = inflateEnd(stream);
+	}
 
 	if(state->fp == NULL)
 		close(state->fd);
 	else
 		fclose(state->fp);
 
+	if (state->buf != NULL)
+		free(state->buf);
 	free(next_out);
 	free(state);
 
 	return ret;
 }
 
-gzFile nx_gzdopen(int fd, const char *mode)
-{
-	return nx_gzopen_(NULL, fd, mode);
-}
-
-gzFile nx_gzopen(const char* path, const char *mode)
+int nx_gzclose(gzFile file)
 {
-	return nx_gzopen_(path, 0, mode);
+	return __gzclose(file, 1);
 }
 
-gzFile nx_gzopen_(const char* path, int fd, const char *mode)
+gzFile __gzopen_(const char* path, int fd, const char *mode, int force_nx)
```

Comment:
Can we get rid of the  trailing `_` and simply use `__gzopen` for the name? That way it's consistent with `__gzwrite` and `__gzclose`.

Reply:
Done.